### PR TITLE
Add discussion How to modernize a setup.py project

### PR DESCRIPTION
**Pull request by sinoroc**
GitHub: refs https://github.com/pypa/packaging.python.org/issues/1334

Preview: https://python-packaging-user-guide--1341.org.readthedocs.build/en/1341/discussions/modernize-setup-py-project/

### CODE REVIEW

File: source/discussions/modernize-setup-py-project.rst [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+.. code:: toml
+
+    [build-system]
+    requires = ["setuptools"]
+    build-backend = "setuptools.build_meta"
+
+
+This is the standardized method of letting :term:`build frontends <Build Frontend>` know
+that :ref:`setuptools` is the :term:`build backend <Build Backend>` for this project.
```
Comment (oscarbenjamin) [inline source/discussions/modernize-setup-py-project.rst:60]: I would note here that adding this enables built-time isolation by default which naturally leads into the next section (build-time dependencies).

Somewhere lower down there could be a section specifically about build isolation and how to disable it when using `pip` or `build`. That section could also say why you might want to disable build isolation e.g. to pin the versions of build requirements.

File: source/discussions/modernize-setup-py-project.rst [hunk 2]
```diff
@@ -0,0 +1,221 @@
+==============================================
+How to modernize a ``setup.py`` based project?
+==============================================
+
+
+Must there be a ``pyproject.toml`` file?
+========================================
+
+Yes. This is strongly recommended.
+The presence of a ``pyproject.toml`` file itself does not bring much.
```
Comment (chrysle) [inline source/discussions/modernize-setup-py-project.rst:10]: Could you use the `:file:` role for these?

File: source/discussions/modernize-setup-py-project.rst [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+For some projects this isolation is unwanted and it can be deactivated as follows:
+
+* ``python -m build --no-isolation``
+* ``python -m install --no-build-isolation``
+
+
+How to handle packaging metadata?
+=================================
+
+All static metadata can be moved to a ``[project]`` table in the ``pyproject.toml`` file.
```
Comment (chrysle) [inline source/discussions/modernize-setup-py-project.rst:121]: Maybe link to the _Declaring project metdata_ spec here?

File: source/discussions/modernize-setup-py-project.rst [hunk 4]
```diff
@@ -0,0 +1,221 @@
+==============================================
```
Comment (chrysle) [inline source/discussions/modernize-setup-py-project.rst:1]:
```suggestion
.. _`modernize setup.py project`:

==============================================
```

File: source/discussions/modernize-setup-py-project.rst (hunk 1 above)
Comment (sinoroc) [reply source/discussions/modernize-setup-py-project.rst]: Thanks, done :)

File: source/discussions/modernize-setup-py-project.rst (hunk 2 above)
Comment (sinoroc) [reply source/discussions/modernize-setup-py-project.rst]: Thanks, done :)

File: source/discussions/modernize-setup-py-project.rst (hunk 3 above)
Comment (sinoroc) [reply source/discussions/modernize-setup-py-project.rst]: Thanks, done :)

File: source/discussions/modernize-setup-py-project.rst (hunk 4 above)
Comment (sinoroc) [reply source/discussions/modernize-setup-py-project.rst:4]: Thanks, done :)

File: source/discussions/modernize-setup-py-project.rst [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+Yes. This is strongly recommended.
+The presence of a :file:`pyproject.toml` file itself does not bring much.
+What is necessary is the ``[build-system]`` table first,
+and then the ``[project]`` table.
+
+.. todo::
+
+    Add note that the presence of pyproject.toml changes the default behavior of pip
+    to use build isolation?
```
Comment (oscarbenjamin) [inline source/discussions/modernize-setup-py-project.rst:20]: This todo is done now.

Comment (sinoroc) [reply source/discussions/modernize-setup-py-project.rst]: thanks!
removed

**Comment by sinoroc** [comment]
Now at https://github.com/pypa/packaging.python.org/pull/1371

**Pull request closed without merging**